### PR TITLE
DOC: Demo incorporating pytest into test classes

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.  Also changes CI GitHub UI photo to illustrate what most users (who are non-owners) will see (i.e. no "squash and merge").

<a href="https://github.com/pandas-dev/pandas/pull/15987#discussion_r111292547">#15987 (comment)</a>

### CODE REVIEW

File: doc/source/contributing.rst [hunk 1]
```diff
@@ -701,6 +701,18 @@ Tests that we have ``parametrized`` are now accessible via the test name, for ex
    test_cool_feature.py::test_dtypes[int8] PASSED
    test_cool_feature.py::test_series[int8] PASSED
 
+Sometimes, it does make sense to bundle test functions together into a single class, either because the test file is testing multiple functions from a single module, and
```
Comment (jreback) [inline doc/source/contributing.rst]: can you make this a ``::note`` ?

Comment (jreback) [inline doc/source/contributing.rst]: actually, let's split this out to a section after 'Writing Tests' and before 'Parametrize' (you may want to move some prose from Parametrize that is related to this as well).

**Comment by jreback** [comment]
nice picture!

File: doc/source/contributing.rst (hunk 1 above)
Comment (gfyoung) [inline doc/source/contributing.rst]: Sure thing.  Done.

**Comment by gfyoung** [comment]
@jreback All comments addressed.  Ready to merge if nothing else.

**Comment by jreback** [comment]
thanks!

**Pull request merged**